src/view/editor/parts/editor-virtual-room.test.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React from 'react';
+
+import { combineReducers, configureStore } from '@reduxjs/toolkit';
+import { screen } from '@testing-library/react';
+
+import { EditorVirtualRoom } from './editor-virtual-room';
+import * as shell from '../../../carbonio-ui-commons/test/mocks/carbonio-shell-ui';
+import { setupTest } from '../../../carbonio-ui-commons/test/test-setup';
+import { reducers } from '../../../store/redux';
+
+const store = configureStore({ reducer: combineReducers(reducers) });
+
+const chatsIntegrationDataTestId = 'chats-room-component';
+const wscIntegrationDataTestId = 'wsc-room-component';
+
+const FakeChatsIntegrationComponent = jest.fn(
+	(): React.JSX.Element => <div data-testid={chatsIntegrationDataTestId} />
+);
+
+const FakeWscIntegrationComponent = jest.fn(
+	(): React.JSX.Element => <div data-testid={wscIntegrationDataTestId} />
+);
+
+describe('Editor virtual rooms', () => {
+	test('If only ChatsRoomSelector is available, it should be rendered', () => {
+		jest
+			.spyOn(shell, 'useIntegratedComponent')
+			.mockReturnValueOnce([FakeChatsIntegrationComponent, true]);
+		jest
```

Comment:
This mock approach relies heavily on the component current implementation. If the order of the calls to the `useIntegrationComponent` will change the component will behave the same, but the tests will fail.
I suggest to use the `mockImplementation` method instead, and, based on the parameter, return a specific value

Reply:
You're right! To do the mock implementation, I had to modify the initial declaration of the `useIntegratedComponent` mock by adding the `id` parameter, which I use to return the correct values. I opened a PR in the submodule https://github.com/zextras/carbonio-ui-commons/pull/152 to add it.

Reply:
Ouch! we forgot to change the type as we did for `useIntegratedFunction` and `useIntegratedAction`. Thanks for fixing it